### ISSUE TITLE
SQLite fails to load due to "java.lang.NoSuchMethodError: Lorg/sqlite/Collation;.xCompare(Ljava/lang/String;Ljava/lang/String;)I]"

### ISSUE THREAD

**Issue by leeN** [SAP/project-fontus#31]
SQLite expects org/sqlite/Collation to have a `xCompare(String, String)` method. This allows to use Java methods to define collations, like this:

```java
Class. forName("org. sqlite. JDBC");
Connection conn = DriverManager. getConnection("jdbc:sqlite:");
 
Collation. create(conn, "REVERSE", new Collation() {
  protected int xCompare(String str1, String str2) {
    return str1.compareTo(str2) * -1;
  }
});
 
conn. createStatement().execute("select c1 from t order by c1 collate REVERSE;");
```

The existence of this method is checked during [loading the sqlite native library](https://github.com/xerial/sqlite-jdbc/blob/master/src/main/java/org/sqlite/core/NativeDB.c#L470). Fontus rewrites this method to be taint-aware, causing the error above.

Slightly curious how we did not hit that before..

**Comment by leeN** [comment]
As we don't have any possibility to detect this at instrumentation time, we have to implement a workaround. Solution would be to special case sqlite in the Instrumenter to add proxies that call the implemented xCompare methods.

**Comment by leeN** [comment]
Fixed in [a61a712](https://github.com/SAP/project-fontus/pull/25/commits/a61a71295d23b1128333611043be2417b45b3cef)
